### PR TITLE
Move first boot to new service

### LINKED ISSUE #210: VM container-disk: VMI does not include tuned-adm kernel-args
When creating the VM container-disks for the traffic-gen and vmi-under-test - a tuned-adm command in [performed](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/main/vms/traffic-gen/scripts/customize-vm#L28). 
This command is supposed to add kernel-args after reboot.
Although the virt-builder used to create the  container-disks is performing reboot - the kernel-args are not added.
Before reboot:
```
# cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1
```
After reboot:
```
# cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
```
diff:
```
skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
```

The absence of this diff may be crucial to the succesful performance of the checkup.

### PR DESCRIPTION
**Pull request by RamLavi**
When creating the vm-under-test and traffic-gen container-disk images, the first-boot script running by the virt-builder command is run on a service created by virt-builder. 

This service is guaranteed to run in the final stages of the boot process [0].
This behavior may create a race with the checkup, that is waiting on the
agentConnected condition [1] being added in order to run the checkup's
executor package. 
The race is happening since the guest-agent service also runs during the final stages of the boot process.

This PR is solving this race by moving the content of the first-boot script of both VMs to a new service that is guaranteed to run before the guest-agent service.

Fixes #210

[0] https://www.libguestfs.org/virt-builder.1.html
[1] https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/b9b6a472fe92583a0db46361289239e0e8d06284/pkg/internal/checkup/checkup.go#L237

### CODE REVIEW

**Comment by RamLavi** [comment]
passes e2e on CNV 4.15 cluster:
```
make test/e2e
podman run --rm \
           --volume /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /home/ralavi/.kube/sno01-cnvqe2-rdu2:/root/.kube:Z,ro \
           --workdir /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/kubeconfig \
           -e TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=dpdk-checkup-ns-1 \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME=dpdk-sriovnetwork-ns-1 \
           -e TRAFFIC_GEN_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest \
           -e VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-vm:latest \
           docker.io/library/golang:1.20.12 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
==============================================================================================================
Random Seed: 1704886098

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.003 seconds]
------------------------------
Execute the checkup Job should complete successfully
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:83
• [394.488 seconds]
------------------------------

Ran 1 of 1 Specs in 394.490 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (394.49s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests394.507s
```

logs show the correct linux cmdline for both VMs:
```
2024/01/10 11:32:45 VMI under test guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
[root@vmi-under-test-pf728 cloud-user]# 
2024/01/10 11:32:47 traffic generator guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
[root@dpdk-traffic-gen-pf728 cloud-user]#
```

**Review by orelmisan — APPROVED** [review]
Thank you

File: vms/vm-under-test/scripts/customize-vm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 systemctl disable NetworkManager-wait-online
 systemctl disable sshd
+set -e
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:20]: rebase mistake. fixed

File: vms/traffic-gen/scripts/customize-vm [hunk 2]
```diff
@@ -27,6 +27,23 @@ disable_services() {
   done
 }
 
+# Install trex package
+install_trex() {
+  TREX_URL=https://trex-tgn.cisco.com/trex/release
+  TREX_VERSION=v3.03
+  TREX_ARCHIVE_NAME=${TREX_VERSION}.tar.gz
+  TREX_DIR=/opt/trex
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply vms/traffic-gen/scripts/customize-vm:35]: fixed

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/3357895769bc39dfb577725d8349feb9d17464b7..a7fbd814e9a1bf919ec95a34be4f7c2a9dfee596): Review fixes

**Review by orelmisan — APPROVED** [review]

**Comment by RamLavi** [comment]
passed e2e with CNV 4.15 cluster:
```
make test/e2e
podman run --rm \
           --volume /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /home/ralavi/.kube/sno01-cnvqe2-rdu2:/root/.kube:Z,ro \
           --workdir /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/kubeconfig \
           -e TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=dpdk-checkup-ns-1 \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME=dpdk-sriovnetwork-ns-1 \
           -e TRAFFIC_GEN_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest \
           -e VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-vm:latest \
           docker.io/library/golang:1.20.12 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
==============================================================================================================
Random Seed: 1704907747

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.002 seconds]
------------------------------
Execute the checkup Job should complete successfully
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:83
• [378.229 seconds]
------------------------------

Ran 1 of 1 Specs in 378.232 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (378.23s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests378.254s
```

kernel Args are correctly updated in the log:
```
2024/01/10 17:33:22 VMI under test guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
[root@vmi-under-test-52qlz cloud-user]# 
2024/01/10 17:33:24 traffic generator guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-365.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 default_hugepagesz=1GB hugepagesz=1G hugepages=1 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 nohz=on nohz_full=2-7 rcu_nocbs=2-7 tuned.non_isolcpus=00000003 intel_pstate=disable nosoftlockup
[root@dpdk-traffic-gen-52qlz cloud-user]# 
```

**Pull request merged**